backend/app/utilities/googleCloud.js

```diff
@@ -0,0 +1,95 @@
+var storage = require('@google-cloud/storage')
+const { logMessage } = require('../utilities/logHelper')
+const moment = require('moment')
+
+const { GCLOUD_KEY_PATH, GCLOUD_BUCKET_NAME } = require('../constants')
+
+
+function uploadFile(srcFileName, destFileName) {
+  let gcs = storage({
+    projectId: process.env.GCLOUD_PROJECT,
+    keyFilename: GCLOUD_KEY_PATH
+  })
+
+  var bucket = gcs.bucket(GCLOUD_BUCKET_NAME)
+
+  // return promise for file upload
+  return new Promise(function(resolve, reject) {
+    logMessage(`++ uploading ${srcFileName} to gcs://${GCLOUD_BUCKET_NAME}/${destFileName}`)
+    bucket.upload(srcFileName, {destination: destFileName}, function (err) {
+      if (!err) {
+        logMessage(`++ successfully uploaded ${srcFileName} to gcs://${GCLOUD_BUCKET_NAME}/${destFileName}`)
+        return resolve()
+      } else {
+        return reject('++ failed to upload to google cloud')
+      }
+    })
+  })
+}
+
+function makeFilePublic(filePath) {
+  let gcs = storage({
+    projectId: process.env.GCLOUD_PROJECT,
+    keyFilename: GCLOUD_KEY_PATH
+  })
+
+  var bucket = gcs.bucket(GCLOUD_BUCKET_NAME)
+
+  // return promise for file upload
+  return new Promise(function(resolve, reject) {
```

Comment:
you don't really need to use a promise constructor here, you could rewrite this as

```
logMesssage(`++ making ${filePath} public`)
return bucket.file(filePath)
  .makePublic()
  .then(() => {
    logMessage(`gs://${GCLOUD_BUCKET_NAME}/${filePath} is now public.`)
  })
  .catch(err => {
    console.error('ERROR:', err)
    throw err
  })
```

Reply:
cool for some reason thought it didn't return a promise before, but seems to be working with this now thanks